src/jdk.javadoc/share/classes/jdk/javadoc/internal/doclets/formats/html/MethodWriterImpl.java

```diff
@@ -293,26 +294,28 @@ protected static void addOverridden(HtmlDocletWriter writer,
      * Adds "implements" information for a method (if appropriate)
      * into a definition list.
      *
-     * @param writer the writer for the method
-     * @param method the method
-     * @param dl     the definition list
+     * @param writer  the writer for the method
+     * @param method  the method
+     * @param methods implemented methods
+     * @param dl      the definition list
      */
     protected static void addImplementsInfo(HtmlDocletWriter writer,
                                             ExecutableElement method,
+                                            Collection<ExecutableElement> methods,
                                             Content dl) {
         Utils utils = writer.utils;
-        if (utils.isStatic(method) || writer.options.noComment()) {
+        if (writer.options.noComment()) {
             return;
         }
         Contents contents = writer.contents;
-        VisibleMemberTable vmt = writer.configuration
-                .getVisibleMemberTable(utils.getEnclosingTypeElement(method));
+        var enclosing = (TypeElement) method.getEnclosingElement();
```

Comment:
Wishlist for Language Model: default methods that return a narrower result type:
```
interface Demo {
   ...>     Element getEnclosingElement(Element e);
   ...>     default TypeElement getEnclosingElement(ExecutableElement ee) {
   ...>         return (TypeElement) getEnclosingElement((Element) ee);
   ...>     }
   ...>     default TypeElement getEnclosingElement(VariableElement ve) {
   ...>         return (TypeElement) getEnclosingElement((Element) ve);
   ...>     }
   ...> }
   ```